web/controllers/account_controller.ex

```diff
@@ -131,14 +128,39 @@ defmodule Healthlocker.AccountController do
   def slam(conn, _params) do
     user_id = conn.assigns.current_user.id
     user = Repo.get!(User, user_id)
-    render conn, "slam.html", user: user
+    changeset = User.connect_slam(%User{})
+    render conn, "slam.html", user: user, changeset: changeset,
+                  action: account_path(conn, :check_slam)
   end
 
-  def slam_help(conn, _params) do
-    render conn, "slam_help.html"
+  def check_slam(conn, %{"user" => %{"Forename" => forename, "Surname" => surname, "NHS_Number" => nhs, "DOB" => dob}}) do
```

Comment:
Generally, when I see non-RESTful actions like this it means that we might have modelled this incorrectly (not always tho!). This feels like it could be the `create` action in a `SlamController` and the `slam` action about could be the `show`?